internal/resources/cluster/resource_cluster.go

```diff
@@ -274,6 +357,66 @@ func resourceClusterCreate(ctx context.Context, d *schema.ResourceData, m interf
 	return append(diags, dataSourceTMCClusterRead(context.WithValue(ctx, contextMethodKey{}, "create"), d, m)...)
 }
 
+type (
+	KubeConfigOption func(*KubeConfig)
+
+	KubeConfig struct {
+		Path string
+		Raw  string
+	}
+)
+
+func WithPath(p string) KubeConfigOption {
+	return func(config *KubeConfig) {
+		config.Path = p
+	}
+}
+
+func WithRaw(r string) KubeConfigOption {
+	return func(config *KubeConfig) {
+		config.Raw = r
+	}
+}
+
+func getK8sClient(opts ...KubeConfigOption) (k8sClient.Client, error) {
+	cfg := &KubeConfig{}
+
+	for _, o := range opts {
+		o(cfg)
+	}
+
+	var (
+		restConfig *rest.Config
+		err        error
+	)
+
+	switch {
+	case cfg.Path != "":
+		restConfig, err = clientcmd.BuildConfigFromFlags("", cfg.Path)
+		if err != nil {
+			return nil, errors.WithMessagef(err, "Invalid kubeconfig file path provided, filepath : %s", cfg.Path)
+		}
+	case cfg.Raw != "":
+		restConfig, err = clientcmd.RESTConfigFromKubeConfig([]byte(cfg.Raw))
+		if err != nil {
+			return nil, errors.WithMessagef(err, "Invalid raw kubeconfig provided.")
+		}
+	}
+
+	if restConfig == nil {
+		return nil, errors.WithMessagef(err, "Kubeconfig not provided.")
+	}
+
+	restConfig.Timeout = 10 * time.Second
+
+	client, err := k8sClient.New(restConfig, k8sClient.Options{})
```

Comment:
check what is the value of the client in the case of the error and you can use it the above if condition check

Reply:
added the check above